wandb/sdk/wandb_settings.py

```diff
@@ -530,6 +530,7 @@ class Settings:
     ignore_globs: Tuple[str]
     init_timeout: float
     is_local: bool
+    job_source: str
```

Comment:
Consider adding a validator to this setting that would check that it is one of the allowed types (look for _validator_... methods for examples on how we do it).